confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverter.java

```diff
@@ -348,6 +352,31 @@ private EntityReference fromSpaceKey(String spaceKey)
         return new EntityReference(convertedSpace, EntityType.SPACE, root);
     }
 
+    /**
+     * Convert an external group ID to a XWiki reference.
+     * @param groupId confluence external group ID
+     * @return serialized XWiki group reference
+     */
+    public Optional<String> convertGroupReference(String groupId)
+    {
+        try {
+            for (long i : context.getConfluencePackage().getGroups()) {
```

Comment:
Well yes I was not sur at 100% about the way I implemented this.

> It may be worth introducing a getGroupById method in ConfluenceXMLPackage.

Do you mean in this class or in the `ConfluenceInputFilterStream` class ?
 
And for the method `getGroupById` do you mean by externalId, as we have an ID and an externalId which is different.

> Have you checked that it works with space exports?

Well, probably no. How to test this ?

Reply:
Should be fixed now.